### ISSUE TITLE
TextInput does not use custom theme unless it is type="search"

### ISSUE THREAD

**Issue by victorfunes** [themesberg/flowbite-react#845]
**Describe the bug**
TextInput only uses custom theme if type is search.

I have created a custom theme for TextInput to follow your Figma designs, to create an input with only a line on the bottom, but it only works using type="search":

```jsx
 <TextInput
      type="search"
      theme={underlineInputTheme}
    />

const underlineInputTheme = {
  field: {
    input: {
      base: "block w-full disabled:cursor-not-allowed disabled:opacity-50 border-0 border-b-2",
      sizes: {
        sm: "py-2 px-0 sm:text-xs",
        md: "py-2.5 px-0 text-sm",
        lg: "py-4 px-0 sm:text-md",
      },
      colors: {
        gray: "bg-gray-50 border-gray-300 text-gray-900 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 focus:border-primary-500 focus:ring-transparent dark:focus:border-primary-500 dark:focus:ring-transparent",
        info: "border-cyan-500 bg-cyan-50 text-cyan-900 placeholder-cyan-700 dark:border-cyan-400 dark:bg-cyan-100 focus:border-cyan-500 focus:ring-transparent dark:focus:border-cyan-500 dark:focus:ring-transparent",
        failure:
          "border-red-500 bg-red-50 text-red-900 placeholder-red-700 dark:border-red-400 dark:bg-red-100 focus:border-red-500 focus:ring-transparent dark:focus:border-red-500 dark:focus:ring-transparent",
        warning:
          "border-yellow-500 bg-yellow-50 text-yellow-900 placeholder-yellow-700 dark:border-yellow-400 dark:bg-yellow-100 focus:border-yellow-500 focus:ring-transparent dark:focus:border-yellow-500 dark:focus:ring-transparent",
        success:
          "border-green-500 bg-green-50 text-green-900 placeholder-green-700 dark:border-green-400 dark:bg-green-100 focus:border-green-500 focus:ring-transparent dark:focus:border-green-500 dark:focus:ring-transparent",
      },
      withAddon: {
        on: "rounded-none",
        off: "rounded-none",
      },
    },
  },
};
```

**To Reproduce**
1. Create a text input component using the custom theme:

```jsx
 <TextInput theme={underlineInputTheme} />
```

3. Click on the input field to write and the style will not be kept (for instance the ring is different).
4. Change the component to type search and the style is working perfectly even when clicking to write:

```jsx
 <TextInput type="search" theme={underlineInputTheme} />
```

**Expected behavior**
Style should be consistent independently of the text input type.

**Screenshots**
<img width="309" alt="TypeSearchScreenshot" src="https://github.com/themesberg/flowbite-react/assets/108006690/c6b9e8a2-cd54-40f5-8c05-2e0962ecdc42">
<img width="296" alt="WithoutTypeScreenshot" src="https://github.com/themesberg/flowbite-react/assets/108006690/8967a344-c502-498c-a9a2-fd053077731e">

**System information:**

- OS: Mac OS
- Browser: Chrome
- Version: 114.0.5735.198 

**Project information:**

- Tailwind: 3.2.4
- Flowbite: 1.6.5
- Flowbite React: 0.4.10
- Type: CRA

**Comment by rluders** [comment]
I would love to understand how `type="search"` would affect the styling. 

https://github.com/themesberg/flowbite-react/blob/65b13e7dd5def51076348bd9e06b6505419167cd/src/components/TextInput/TextInput.tsx#L87-95

The code doesn't seem to have anything that may cause this behavior. 🤔

**Comment by rluders** [comment]
BTW, I was applying the theme at the [storybook](https://flowbite-react-git-storybook-themesberg.vercel.app/?path=/story/components-textinput--default) and I didn't see the issue happening there.

Maybe it requires some better investigation.

**Comment by victorfunes** [comment]
I have exactly the same behaviour in storybook:

Input not selected:
<img width="1061" alt="Screenshot 2023-07-08 at 09 38 09" src="https://github.com/themesberg/flowbite-react/assets/108006690/d1e40737-7abf-4427-ad6d-4b80cffdcfe7">

Input selected:
<img width="1061" alt="Screenshot 2023-07-08 at 09 38 27" src="https://github.com/themesberg/flowbite-react/assets/108006690/2ffd7fdf-db7f-40d7-bc54-c65ed9b867d7">

**Comment by rluders** [comment]
All right, I noticed that the `:focus-visible` is triggering the ring effect. Try to use it instead: https://tailwindcss.com/docs/hover-focus-and-other-states#pseudo-class-reference

**Comment by victorfunes** [comment]
Thank you for your super quick response! I am not sure if I undestand what you mean.
I have added: focus-visible:ring-transparent dark:focus-visible:ring-transparent but the behaviour is still the same.
Did I miss something?

`{"field":{"input":{"base":"block w-full disabled:cursor-not-allowed disabled:opacity-50 border-0 border-b-2","sizes":{"sm":"py-2 px-0 sm:text-xs","md":"py-2.5 px-0 text-sm","lg":"py-4 px-0 sm:text-md"},"colors":{"gray":"bg-gray-50 border-gray-300 text-gray-900 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 focus:border-primary-500 focus:ring-transparent dark:focus:border-primary-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent","info":"border-cyan-500 bg-cyan-50 text-cyan-900 placeholder-cyan-700 dark:border-cyan-400 dark:bg-cyan-100 focus:border-cyan-500 focus:ring-transparent dark:focus:border-cyan-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent","failure":"border-red-500 bg-red-50 text-red-900 placeholder-red-700 dark:border-red-400 dark:bg-red-100 focus:border-red-500 focus:ring-transparent dark:focus:border-red-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent","warning":"border-yellow-500 bg-yellow-50 text-yellow-900 placeholder-yellow-700 dark:border-yellow-400 dark:bg-yellow-100 focus:border-yellow-500 focus:ring-transparent dark:focus:border-yellow-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent","success":"border-green-500 bg-green-50 text-green-900 placeholder-green-700 dark:border-green-400 dark:bg-green-100 focus:border-green-500 focus:ring-transparent dark:focus:border-green-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent"},"withAddon":{"on":"rounded-none","off":"rounded-none"}}}}`

**Comment by rluders** [comment]
@victorfunes I think that the `focus-visible:` will not be at the Storybook (it doesn't actually have all the TailwindCSS classes), but test on your project, it should world as expected. At least, I tested here and I saw no issue.

**Comment by victorfunes** [comment]
I did it first in my project, and after failing I tested it in storybook and the behaviour is still the same on both.
I have another issue with all my dropdown list not closing after the second time clicking in an item in my project but I am not reporting it as it works in storybook.
Based on this I try to test it in both places before reporting you any issue.

**Comment by rluders** [comment]
I'm using the code that we have in the documentation + your latest custom theme for the input. Note that the first input has customization, and the second still uses the standard style.

https://github.com/themesberg/flowbite-react/assets/191027/a755817c-f245-4fd3-9277-272ec9fbe5ef

**Comment by rluders** [comment]
```jsx
<form className="flex max-w-md flex-col gap-4">
  <div>
    <div className="mb-2 block">
      <Label htmlFor="email1" value="Your email" />
    </div>
    <TextInput
      theme={{
        field: {
          input: {
            base: 'block w-full disabled:cursor-not-allowed disabled:opacity-50 border-0 border-b-2',
            sizes: {
              sm: 'py-2 px-0 sm:text-xs',
              md: 'py-2.5 px-0 text-sm',
              lg: 'py-4 px-0 sm:text-md',
            },
            colors: {
              gray: 'bg-gray-50 border-gray-300 text-gray-900 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 focus:border-primary-500 focus:ring-transparent dark:focus:border-primary-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent',
              info: 'border-cyan-500 bg-cyan-50 text-cyan-900 placeholder-cyan-700 dark:border-cyan-400 dark:bg-cyan-100 focus:border-cyan-500 focus:ring-transparent dark:focus:border-cyan-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent',
              failure:
                'border-red-500 bg-red-50 text-red-900 placeholder-red-700 dark:border-red-400 dark:bg-red-100 focus:border-red-500 focus:ring-transparent dark:focus:border-red-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent',
              warning:
                'border-yellow-500 bg-yellow-50 text-yellow-900 placeholder-yellow-700 dark:border-yellow-400 dark:bg-yellow-100 focus:border-yellow-500 focus:ring-transparent dark:focus:border-yellow-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent',
              success:
                'border-green-500 bg-green-50 text-green-900 placeholder-green-700 dark:border-green-400 dark:bg-green-100 focus:border-green-500 focus:ring-transparent dark:focus:border-green-500 dark:focus:ring-transparent focus-visible:ring-transparent dark:focus-visible:ring-transparent',
            },
            withAddon: {
              on: 'rounded-none',
              off: 'rounded-none',
            },
          },
        },
      }}
      id="email1"
      type="email"
      placeholder="name@flowbite.com"
      required
    />
  </div>
  <div>
    <div className="mb-2 block">
      <Label htmlFor="password1" value="Your password" />
    </div>
    <TextInput id="password1" type="password" required />
  </div>
  <div className="flex items-center gap-2">
    <Checkbox id="remember" />
    <Label htmlFor="remember">Remember me</Label>
  </div>
  <Button type="submit">Submit</Button>
</form>
```

**Comment by victorfunes** [comment]
Ok, now I see where is the difference:
If I specify the type, the component works (whatever the type is).
If I delete the type and don't specify it, then it doesn't work.

**Comment by rluders** [comment]
After careful consideration, I have determined that the issue regarding the omission of the `type=` attribute indeed leads to a malfunction. Nevertheless, specifying `type="text"` or any other valid value ensures proper functionality. Given the straightforward and uncomplicated nature of the workaround, coupled with the lack of understanding regarding the underlying cause, I have decided to label this issue as "won't fix" and proceed with its closure.
